R/use_github_file.R

```diff
@@ -110,7 +115,11 @@ read_github_file <- function(repo_spec, path, ref = NULL, host = NULL) {
     .destfile = tf,
     .accept = "application/vnd.github.v3.raw"
   )
-  read_utf8(tf)
+  if (is.null(res$error)) {
+    read_utf8(tf)
+  } else {
+    structure(NA_character_, error = res$error)
+  }
```

Comment:
My basic goal is to make this more of a true helper and to let the caller decide what to do about a failure to retrieve file contents.

We need that because, for example, there's no guarantee that `NEWS.md` exists when drafting a release (of course, we think there should be, but it would be nice to not error when `NEWS.md` can't be found).

Reply:
This doesn't feel quite right to me, because generally errors should bubble up by default. If you don't want to error, _then_ you need to do something special.

Why not use `tryCatch()` with the existing conditions that gh/httr/httr2 throws?

Reply:
OK I'm redoing this PR.

Is this difference in the error condition intentional in the latest gh release? We seem to have lost a lot of information (e.g. the HTTP status code) and it's not marked as an `"rlang_error"` or `"error"` at all.

``` r
# pak::pak("r-lib/gh@v1.3.1")
devtools::load_all("~/rrr/usethis")
#> ℹ Loading usethis
packageVersion("gh")
#> [1] '1.3.1'
x <- rlang::catch_cnd(read_github_file("r-lib/usethis", path = "NOPE"))
class(x)
#> [1] "github_error"   "http_error_404" "rlang_error"    "error"         
#> [5] "condition"
```

``` r
devtools::load_all("~/rrr/usethis")
#> ℹ Loading usethis
packageVersion("gh")
#> [1] '1.4.0'
x <- rlang::catch_cnd(read_github_file("r-lib/usethis", path = "NOPE"))
class(x)
#> [1] "httr2_fetch" "condition"
```

Reply:
Do you mind filing an issue in gh, and then I'll tackle it tomorrow?

Reply:
It's a weird situation:

https://github.com/r-lib/gh/issues/179

Reply:
OK I have un-confused myself about the gh error classes (it's in the issue).

Reply:
Oooh that makes sense.